BHoM_Engine/Create/Type/Type.cs

```diff
@@ -51,6 +51,18 @@ public static Type Type(string name, bool silent = false)
 
```

Comment:
Think this is the clear pinch point in the deserialisation process.

Did some profiling, and some testing on a set of random Bars, and from that, the majority of the time for the de-serialisation was highlighted to be the linq query of creating a new dictionary in the BHoMTypeDictionary method:

https://github.com/BHoM/BHoM_Engine/blob/27983e8f22b39b98778cfdb4b6ee5e46af56f67c/BHoM_Engine/Query/TypeDictionary.cs#L39-L42

if we simply skip that method here, and directly just access the underlying concurrent dictionary from the Global variable, by changing the line here to:

```
            ConcurrentDictionary<string, List<Type>> typeDictionary = Global.BHoMTypeDictionary;
```

we get a speedup (for my test case) that brings the de-serialisation to the same level as the old system. I ran a few tests, and they both compare, with some cases favouring the old, and some the new. 

One option if we dont want to directly access the concurrent dictionary here is to change the return type of the BHoMTypeDictionary to `IReadOnlyDictionary` and simply return the concurrent dictionary. Not sure if that could have other implications elsewhere though, as on BHoM_UI and similar.